### PR TITLE
Support K8s 1.18 IngressClass

### PR DESCRIPTION
**Pull request by howardjohn**

### CODE REVIEW

**Comment by linsun** [comment]
@howardjohn is the ingressclass related to ingress v2 API that has gateway and routerules?

**Comment by howardjohn** [comment]
No it's a different API, this is for Ingress which is the old one but they
are moving it to v1 soon

**Comment by linsun** [comment]
Thanks, sorry folks were confused it with the new ingress v2 API.

File: pilot/pkg/config/kube/ingress/controller.go [hunk 1]
```diff
@@ -18,13 +18,18 @@ package ingress
 
 import (
 	"errors"
+	"fmt"
 	"reflect"
 	"time"
 
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/informers/networking/v1beta1"
+
```
Comment (hzxuzhonghu) [inline pilot/pkg/config/kube/ingress/controller.go:27]: remove

File: pilot/pkg/config/kube/ingress/conversion.go [hunk 2]
```diff
@@ -35,6 +35,10 @@ import (
 	"istio.io/istio/pkg/config/protocol"
 )
 
+const (
+	IstioIngressController = "istio.io/ingress-controller"
```
Comment (hzxuzhonghu) [inline pilot/pkg/config/kube/ingress/conversion.go:39]: Is this a new value?

Comment (howardjohn) [reply pilot/pkg/config/kube/ingress/conversion.go:39]: Yes

Comment (hzxuzhonghu) [reply pilot/pkg/config/kube/ingress/conversion.go:39]: Well, should update document

**Review by hzxuzhonghu — APPROVED** [review]
I understand there should be only one IngressClass for a controller. Though k8s does not restrict this.

**Comment by howardjohn** [comment]
documentation is in https://github.com/istio/istio.io/pull/7012

**Comment by howardjohn** [comment]
You can have multiple ingress class for the controller to allow specifying
different options for them, for example, I think. However we don't have any
customization for Ingress so it's not needed for us

**Comment by howardjohn** [comment]
ping @istio/wg-environments-maintainers for role addition

**Review by morvencao — APPROVED** [review]

**Pull request merged**